localstate/localstate.go

```diff
@@ -1,19 +1,43 @@
 package localstate
 
 import (
+	"context"
 	"encoding/json"
+	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 
 	"github.com/docker/docker/pkg/ioutils"
 	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
 )
 
-const refsDir = "refs"
+const (
+	refsDir  = "refs"
+	groupDir = "__group__"
+)
 
 type State struct {
-	LocalPath      string
+	// Target is the name of the invoked target (default if empty)
+	Target string
+	// LocalPath is the absolute path to the context
+	LocalPath string
+	// DockerfilePath is the absolute path to the Dockerfile
 	DockerfilePath string
+	// GroupID is the ID of the state group that this ref belongs to
+	GroupID string `json:",omitempty"`
+}
+
+type StateGroup struct {
+	// Definition is the raw representation of the group (bake definition)
+	Definition []byte
+	// Targets are the targets invoked
+	Targets []string `json:",omitempty"`
+	// Inputs are the user inputs (bake overrides)
```

Comment:
So no variables? Ok by me to leave this for follow-up but comments seem to suggest that this was already added.

Reply:
Will look in follow-up. It's a bit tricky as the `matrix` feature hacks around variables so we can't just pick them.

Reply:
I think if you want the values that the variables are evaluated to, you'd need to modify `hclparser.Parse` to return that information. I don't think there's a way to extract the vars/funcs defined in the HCL today :cry: 

Apologies, the code around there is a total mess - happy to help out :smile:

Reply:
> you'd need to modify `hclparser.Parse` to return that information.

Yes indeed, I have started to look at it and how I could use context evaluation to extract vanilla variables. Will open a PR and let you know, thx!